backend/agents/goal_agent.py

```diff
@@ -0,0 +1,52 @@
+import functools
+import logging
+from utils import Config
+from utils import graph_db_utils
+from utils.llm import call_model_with_tools
+
+logger = logging.getLogger(__name__)
+config = Config()
+
+tools = [
+    {
+        "type": "function",
+        "function": {
+            "description": "Get a goal and its description from the user",
+            "name": "create_goal",
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "goal_name": {
+                        "type": "string",
+                        "description": "As a string, A single title keyword to summarise the goal eg. mortgage",
+                    },
+                    "goal_description": {
+                        "type": "string",
+                        "description": "A string to summarise the goal",
+                    },
+                },
+                "required": ["goal_name"],
+            },
+        },
+    }
+]
+ 
+names_to_functions = {
+    'create_goal': functools.partial(graph_db_utils.create_goal),
```

Comment:
this looks cool, what is `functools.partial` doing here?

Reply:
Where the string `create_goal` appears, it assigns the function as the dictionary value. The function can then be called. You can see this being called on line 52